### PR TITLE
Better columns number settings

### PR DESCRIPTION
**Pull request by catsuryuu**
https://github.com/codilime/veles/issues/260

### CODE REVIEW

File: include/ui/hexedit.h [hunk 1]
```diff
@@ -42,13 +42,15 @@ class HexEdit : public QAbstractScrollArea {
           QItemSelectionModel *selectionModel = nullptr, QWidget *parent = 0);
   /** Mark bytes as selected and optionally scroll screen to make these bytes visible */
   void setSelection(qint64 start, qint64 size, bool set_visible = false);
+  int getBytesPerRow() { return bytesPerRow_; }
```
Comment (mkow) [inline include/ui/hexedit.h]: Add const

File: include/ui/hexedit.h [hunk 2]
```diff
@@ -42,13 +42,15 @@ class HexEdit : public QAbstractScrollArea {
...
   /** Sets how many bytes should be displayed in the single hex row or optionaly
    *  turn on automatic mode which will adjust bytes per row to window size */
-  void setBytesPerRow(int bytesCount, bool automatic = false);
+  void setBytesPerRow(int bytesCount, bool automatic);
```
Comment (mkow) [inline include/ui/hexedit.h]: Please use Google Code Style in new code - should be `bytes_count`.

File: include/ui/hexeditwidget.h [hunk 3]
```diff
@@ -26,6 +26,7 @@
 #include <QToolButton>
 #include <QMainWindow>
 #include <QSharedPointer>
+#include <QWidgetAction>
```
Comment (mkow) [inline include/ui/hexeditwidget.h]: Please add newline after this section of includes (yes, it was wrong before ;) ).

File: include/ui/hexeditwidget.h [hunk 4]
```diff
@@ -57,6 +58,8 @@ class HexEditWidget : public View {
   QAction* showVisualizationAction() {return visualization_act_;}
   QAction* showHexEditAction() {return show_hex_edit_act_;}
   QAction* showNodeTreeAction() {return show_node_tree_act_;}
+  QAction* addColumnAction() {return add_column_act_;}
+  QAction* removeColumnAction() {return remove_column_act_;}
```
Comment (mkow) [inline include/ui/hexeditwidget.h]: add consts

File: include/util/settings/shortcuts.h [hunk 5]
```diff
@@ -102,6 +102,8 @@ enum ShortcutType : uint32_t {
   DOCK_CLOSE = 67,
   SWITCH_TAB_NEXT = 68,
   SWITCH_TAB_PREV = 69,
+  HEX_ADD_COLUMN = 72,
+  HEX_REMOVE_COLUMN = 73,
```
Comment (mkow) [inline include/util/settings/shortcuts.h]: This still conflicts on merge, please rebase to newest master

File: include/ui/hexeditwidget.h (hunk 4 above)
Comment (catsuryuu) [inline include/ui/hexeditwidget.h]: Only here? (These two lines?)

Comment (mkow) [inline include/ui/hexeditwidget.h]: You can also fix the rest of these getter if you want ;)

File: include/ui/hexedit.h [hunk 6]
```diff
@@ -42,9 +42,11 @@ class HexEdit : public QAbstractScrollArea {
           QItemSelectionModel *selectionModel = nullptr, QWidget *parent = 0);
   /** Mark bytes as selected and optionally scroll screen to make these bytes visible */
   void setSelection(qint64 start, qint64 size, bool set_visible = false);
+  const int getBytesPerRow() { return bytesPerRow_; }
```
Comment (mkow) [inline include/ui/hexedit.h]: wrong const :P

File: src/ui/hexeditwidget.cc [hunk 7]
```diff
@@ -364,6 +394,24 @@ void HexEditWidget::editStateChanged(bool has_changes, bool has_undo) {
   undo_act_->setEnabled(has_undo);
 }
 
+void HexEditWidget::addColumn() {
+  int bytesPerRow = hex_edit_->getBytesPerRow();
```
Comment (mkow) [inline src/ui/hexeditwidget.cc]: should be `bytes_per_row`

File: src/ui/hexeditwidget.cc [hunk 8]
```diff
@@ -364,6 +394,24 @@ void HexEditWidget::editStateChanged(bool has_changes, bool has_undo) {
...
+  if (bytesPerRow >= 1024) {
+    return;
+  }
+  hex_edit_->setBytesPerRow(bytesPerRow + 1, false);
+  auto_resize_checkbox_->setChecked(false);
+}
+
+void HexEditWidget::removeColumn() {
+  int bytesPerRow = hex_edit_->getBytesPerRow();
```
Comment (mkow) [inline src/ui/hexeditwidget.cc]: ditto

**Pull request merged**